src/univariate/continuous/skewexponentialpower.jl

```diff
@@ -0,0 +1,115 @@
+"""
+    SkewExponentialPower(μ, σ, p, α)
+
+The *Skewed exponential power distribution*, with location `μ`, scale `σ`, shape `p`, and skewness `α`
+has the probability density function [1]
+```math
+f(x; \\mu, \\sigma, p, \\alpha) =
+\\begin{cases}
+\\frac{1}{\\sigma 2p^{1/p}\\Gamma(1+1/p)} \\exp \\left\\{ - \\frac{1}{2p}\\Big| \\frac{x-\\mu}{\\alpha \\sigma} \\Big|^p \\right\\}, & \\text{if } x \\leq \\mu \\\\
+\\frac{1}{\\sigma 2p^{1/p}\\Gamma(1+1/p)} \\exp \\left\\{ - \\frac{1}{2p}\\Big| \\frac{x-\\mu}{(1-\\alpha) \\sigma} \\Big|^p \\right\\}, & \\text{if } x > \\mu
+\\end{cases}.
+```
+The Skewed exponential power distribution (SEPD) incorporates the laplace (``p=1, \\alpha=0.5``),
+normal (``p=2, \\alpha=0.5``), uniform (``p\\rightarrow \\infty, \\alpha=0.5``), asymmetric laplace (``p=1``), skew normal (``p=2``),
+and exponential power distribution (``\\alpha = 0.5``) as special cases.
+
+[1] Zhy, D. and V. Zinde-Walsh (2009). Properties and estimation of asymmetric exponential power distribution. _Journal of econometrics_, 148(1):86-96, 2009.
+
+```julia
+SkewExponentialPower()            # SEPD with shape 2, scale 1, location 0, and skewness 0.5 (the standard normal distribution)
+SkewExponentialPower(μ, σ, p, α)  # SEPD with location μ, scale σ, shape p, and skewness α
+SkewExponentialPower(μ, σ, p)     # SEPD with location μ, scale σ, shape p, and skewness 0.5 (the exponential power distribution)
+SkewExponentialPower(μ, σ)        # SEPD with location μ, scale σ, shape 2, and skewness 0.5 (the normal distribution)
+SkewExponentialPower(μ)           # SEPD with location μ, scale 1, shape 2, and skewness 0.5 (the normal distribution)
+
+params(d)       # Get the parameters, i.e. (μ, σ, p, α)
+shape(d)        # Get the shape parameter, i.e. p
+location(d)     # Get the location parameter, i.e. μ
+scale(d)        # Get the scale parameter, i.e. σ
+```
+"""
+struct SkewExponentialPower{T <: Real} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
+    p::T
+    α::T
+    SkewExponentialPower{T}(μ::T, σ::T, p::T, α::T) where {T} = new{T}(μ, σ, p, α)
+end
+
+function SkewExponentialPower(µ::T, σ::T, p::T, α::T; check_args=true) where {T <: Real}
+    check_args && @check_args(SkewExponentialPower, σ > zero(σ))
+    check_args && @check_args(SkewExponentialPower, p > zero(p))
+    check_args && @check_args(SkewExponentialPower, α > zero(α) && α < one(α))
+    return SkewExponentialPower{T}(µ, σ, p, α)
+end
+
+SkewExponentialPower(μ::Real, σ::Real, p::Real, α::Real) = SkewExponentialPower(promote(μ, σ, p, α)...)
+SkewExponentialPower(μ::Real, σ::Real, p::Real) = SkewExponentialPower(promote(μ, σ, p, 0.5)...)
+SkewExponentialPower(μ::Real, σ::Real) = SkewExponentialPower(promote(μ, σ, 2., 0.5)...)
+SkewExponentialPower(μ::Real) = SkewExponentialPower(promote(μ, 1., 2., 0.5)...)
+SkewExponentialPower() = SkewExponentialPower(0., 1., 2., 0.5)
```

Comment:
A bit simpler, with less invasive promotions (useful e.g. when working with `Float32`), and the possibility to specify `check_args`:
```suggestion
function SkewExponentialPower(μ::Real=0, σ::Real=1, p::Real=2, α::Real=1//2; kwargs...)
    return SkewExponentialPower(promote(μ, σ, p, α)...; kwargs...)
end
```